csw-location/src/main/scala/csw/services/location/internal/Networks.scala

```diff
@@ -4,68 +4,56 @@ import java.net.{Inet6Address, InetAddress, NetworkInterface}
 
 /**
-  * Picks an appropriate ipV4 address to register
+  * Picks an appropriate ipv4 address to register with CRDT using the NetworkInterfaceProvider
   *
-  * @param interfaceName   The name of network interface where akka cluster will be formed and an ipV4 address from that
-  *                        network interface will be picked.
-  * @param networkProvider A custom `NetworkInterfaceProvider` which manages network interfaces and their ip addresses
+  * @param interfaceName Provide the name of network interface where csw cluster is running
   */
 class Networks(interfaceName: String, networkProvider: NetworkInterfaceProvider) {
 
   /**
-    * Picks an appropriate ipV4 address to register. Default `NetworkInterfaceProvider` will be created to manage network
-    * interfaces and their ip addresses
-    *
-    * @param interfaceName The name of network interface where akka cluster will be formed and an ip address from that network
-    *                      interface will be picked
+    * Picks an appropriate ipv4 address from the network interface provided
     */
   def this(interfaceName: String) = this(interfaceName, new NetworkInterfaceProvider())
 
   /**
-    * Picks an appropriate ipV4 address to register. Default `NetworkInterfaceProvider` will be created to manage network
-    * interfaces and their ip addresses. Since no `interfaceName` is given all network interfaces will be picked to find
-    * appropriate ipV4 address
+    * Picks an appropriate ipv4 address. Since no specific network interface is provided, the first available interface will be
+    * taken to pick address
     */
   def this() = this("")
 
   /**
-    * Returns ipv4 host address
-    */
+    * Gives the ipv4 host address
+     */
   def hostname(): String = getIpv4Address().getHostAddress
 
   /**
-    * Picks an ipV4 address that is not a loopback address. If the network interface name is provided then the address is
-    * picked from that interface name, else for all network interfaces, all ipV4 addresses which are not loopback are taken
-    * and the first one is picked.
-    *
-    * @return An ipv4 address
+    * Gives the non-loopback, ipv4 address for the given network interface. If no interface name is provided then the address mapped
+    * to the first available interface is chosen.
     */
   def getIpv4Address(): InetAddress = all()
     .sortBy(_._1)
     .find(pair => isIpv4(pair._2))
     .getOrElse((0, InetAddress.getLocalHost))
     ._2
 
+  // Check if the given InetAddress is not a loopback address and is a ipv4 address
   private def isIpv4(addr: InetAddress): Boolean = {
     // Don't use ipv6 addresses yet, since it seems to not be working with the current akka version
     !addr.isLoopbackAddress && !addr.isInstanceOf[Inet6Address]
   }
 
+  // Prepare a tuple of interface index to InetAddress
   private def all(): Seq[(Int, InetAddress)] = {
     for {
       tuple <- getNetworkInterfaceList()
       a <- tuple._2
```

Comment:
Apart from docs: This code could be better, tuples can be directly pattern watch within for, no need for positional methods.